### PR TITLE
Consumption

### PR DESCRIPTION
**Pull request by seanyles**
Consume food. Will automatically prioritize stocks with earlier expiration dates. Just deletes the stock if you go over

### CODE REVIEW

File: src/main/java/com/app/eece4792mealmaster/dto/MealFoodDto.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.app.eece4792mealmaster.dto;
+
+public class MealFoodDto {
```
Comment (mavalos1) [inline src/main/java/com/app/eece4792mealmaster/dto/MealFoodDto.java:3]: This and MealRecipeDto are identical

File: src/main/java/com/app/eece4792mealmaster/services/MealService.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    @Autowired
+    FoodStockRepository foodStockRepository;
+
+    @Autowired
+    StockItemRepository stockItemRepository;
+
+    public void consume(Long userId, MealDto mealData) {
+        Optional<User> oUser = userRepository.findById(userId);
+        if (!oUser.isPresent()) return; // this should never happen
```
Comment (mavalos1) [inline src/main/java/com/app/eece4792mealmaster/services/MealService.java:36]: should throw an error

File: src/main/java/com/app/eece4792mealmaster/services/MealService.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            Optional<Recipe> oRecipe = recipeRepository.findById(cRecipe.getId());
+            if (!oRecipe.isPresent()) return; // should be proper error handling but relying on FE to not fuck up here
+            Collection<RecipeIngredient> cRecipeIngredients = oRecipe.get().getRecipeIngredients();
+            for (RecipeIngredient ingredient : cRecipeIngredients) {
+                Long foodId = ingredient.getIngredient().getId();
+                if (toConsume.containsKey(foodId)) {
+                    toConsume.replace(foodId, toConsume.get(foodId) + cRecipe.getServings() * ingredient.getServings());
+                } else {
+                    toConsume.put(foodId, cRecipe.getServings() * ingredient.getServings());
+                }
```
Comment (mavalos1) [inline src/main/java/com/app/eece4792mealmaster/services/MealService.java:60]: should be a reuseable function

File: src/main/java/com/app/eece4792mealmaster/dto/MealFoodDto.java (hunk 1 above)
Comment (seanyles) [reply src/main/java/com/app/eece4792mealmaster/dto/MealFoodDto.java:3]: Yeah but one is for food and one is for recipe

**Comment by mavalos1** [comment]
rebasing and merging

**Pull request merged**